File: code/common/LibHNLobby/HNLobby/Shop/ProductInfo.cpp
#include "HNLobby/Shop/ProductInfo.h"
#include "json/rapidjson.h"
#include "json/document.h"
#include "json/stringbuffer.h"
#include "json/writer.h"
#include "cocos2d.h"
USING_NS_CC;

namespace HN {
std::string PRODUCT_INFO::serialize() {
    rapidjson::Document doc;
    doc.SetObject();
    rapidjson::Document::AllocatorType& allocator = doc.GetAllocator();
    rapidjson::Value object(rapidjson::kObjectType);



    //object.AddMember(rapidjson::Value("identifier", allocator), rapidjson::Value(identifier.c_str(), allocator), allocator);
    //object.AddMember(rapidjson::Value("number", allocator), rapidjson::Value(number), allocator);
    //object.AddMember(rapidjson::Value("price", allocator), rapidjson::Value(price), allocator);
    //object.AddMember(rapidjson::Value("payType", allocator), rapidjson::Value(payType), allocator);
    //object.AddMember(rapidjson::Value("orderID", allocator), rapidjson::Value(orderID.c_str(), allocator), allocator);
    //object.AddMember(rapidjson::Value("IP", allocator), rapidjson::Value(IP.c_str(), allocator), allocator);

    //rapidjson::StringBuffer buffer;
    //rapidjson::Writer<rapidjson::StringBuffer> write(buffer);
    //object.Accept(write);
    //return buffer.GetString();

    std::string tmp;
    tmp += "{\"identifier\":\"" + identifier + "\",";
    tmp += "\"number\":" + StringUtils::format("%ld", number) + ",";
    tmp += "\"price\":" + StringUtils::format("%f", price) + ",";
    tmp += "\"payType\":" + StringUtils::format("%d", payType) + ",";
    tmp += "\"orderID\":\"" + StringUtils::format("%s", orderID.c_str()) + "\",";
    tmp += "\"IP\":\"" + StringUtils::format("%s", IP.c_str()) + "\"}";
    return tmp;
}

void PRODUCT_INFO::deserialize(std::string data) {
    rapidjson::Document doc;
    doc.Parse<rapidjson::kParseDefaultFlags>(data.c_str());
    if (!doc.HasParseError()) {
        if (doc.IsObject() && doc.HasMember("status")) {
            identifier = doc["identifier"].GetString();
            number = doc["number"].GetInt64();
            price = doc["price"].GetDouble();
            payType = doc["payType"].GetInt();
            orderID = doc["orderID"].GetString();
            IP = doc["IP"].GetString();
        }
    }
}
}

File: code/common/LibHNMarket/platform/android/HNPay/src/com/hn/pay/base/ProductInfo.java
package com.hn.pay.base;

import org.json.JSONException;
import org.json.JSONObject;

public class ProductInfo {
	// 购买数量
	public long number = 0;
	// 商品价格
	public double price = 0;
	// 商品ID
	public String identifier = null;
	// 支付类型 sdk
	public int payType = -1;
	// 订单号
	public String orderID = null;
	// 发送订单的IP
	public String IP = null;

	public ProductInfo() {
	}

	public ProductInfo(long number, double price, String identifier, int payType, String orderID, String IP) {
		// 购买数量
		this.number = number;
		// 商品价格
		this.price = price;
		// 商品ID
		this.identifier = identifier;
		// 支付类型 sdk
		this.payType = payType;
		// 订单号
		this.orderID = orderID;
		this.IP = IP;
	}

	public String serialize() {
		return "";
	}

	public void deserialize(String data) {
		try {
			JSONObject json = new JSONObject(data);
			this.number = json.getInt("number");
			this.price = json.getDouble("price");
			this.identifier = json.getString("identifier");
			this.payType = json.getInt("payType");
			this.orderID = json.getString("orderID");
			this.IP = json.getString("IP");
		} catch (JSONException e) {
			e.printStackTrace();
		}
	}
}
